code/game/machinery/bots/buttbot.dm

```diff
@@ -37,16 +37,17 @@ Here it is: Buttbot.
 /obj/machinery/bot/buttbot/proc/speak(var/message)
 	if((!src.on) || (!message))
 		return
-	for(var/mob/O in hearers(src, null))
-		O.show_message("<b>[src]</b> beeps, '[message]'")
+	for(var/mob/O in hearers(world.view, src))
+		to_chat(O, "<b>[src]</b> beeps, \"[message]\"")
 	return
 
 
 /obj/machinery/bot/buttbot/Hear(var/datum/speech/speech, var/rendered_speech="")
 	set waitfor = 0 //Buttbots speaking should be queued after the original speech completes
-	if(prob(buttchance) && !findtext(speech.message,"butt"))
+	var/message = speech.message // need to save this here before sleeping: the speech datum may be destroyed in the meantime
+	if(prob(buttchance) && !findtext(message,"butt"))
```

Comment:
Here's an alternative.
```
if(prob(buttchance)
	if(istype(speech.speaker, /obj/machinery/bot/buttbot) && speech.speaker != src)
		explode()
		return
	else
		//the rest of the proc
```

tl;dr if a buttbot attempts to add butts to a message said by another buttbot it'll explode

you can still shit up the radio with a buttbot, but you can't spam the radio with buttbots for too long

Reply:
you can still shit up comms for everyone with 1 buttbot and N radios

Reply:
People who do that should honestly just get banned for nonatag griefing.

Reply:
keep radio chatter disabled for buttbots, give them their own internal radio you can enable/disable with the click that makes them fart

Reply:
Honestly I don't think is a good solution
They'd explode pretty quickly since it's an 80% chance to butt

Reply:
You missed the istype.

Reply:
Still insane though